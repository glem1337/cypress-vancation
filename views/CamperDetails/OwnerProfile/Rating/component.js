import classNames from 'classnames';
import PropTypes from 'prop-types';

const Rating = ({
 rating, isHigh, isNormal, isMiddle, isLow,
}) => (
  <>
    <div
      className={classNames('profile-user-card__photo-rating-icon', {
        'background-green-300': isHigh,
        'background-yellow-1000': isMiddle,
        'background-red-1000': isLow,
      })}
    >
      <img src="/images/Like - White.svg" alt="" />
    </div>
    <p
      className={classNames('mr-16 text-subheader font-700', {
        'in-green-300': isHigh,
        'in-green-1000': isNormal,
        'in-yellow-1000': isMiddle,
        'in-red-1000': isLow,
      })}
    >
      {rating}
      %
    </p>
  </>
);

Rating.propTypes = {
  isHigh: PropTypes.bool.isRequired,
  isNormal: PropTypes.bool.isRequired,
  isMiddle: PropTypes.bool.isRequired,
  isLow: PropTypes.bool.isRequired,
  rating: PropTypes.number.isRequired,
};

export default Rating;
