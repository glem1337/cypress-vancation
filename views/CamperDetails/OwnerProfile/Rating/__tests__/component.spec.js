import { shallow } from 'enzyme';

import Rating from '../component';

describe('Rating component tests', () => {
  const props = {
    rating: 80,
    isHigh: false,
    isNormal: true,
    isMiddle: false,
    isLow: false,
  };

  const component = shallow(<Rating {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
