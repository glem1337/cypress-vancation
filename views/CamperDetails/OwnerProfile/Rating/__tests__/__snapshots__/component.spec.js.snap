// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component tests matches snapshot 1`] = `
<Fragment>
  <div
    className="profile-user-card__photo-rating-icon"
  >
    <img
      alt=""
      src="/images/Like - White.svg"
    />
  </div>
  <p
    className="mr-16 text-subheader font-700 in-green-1000"
  >
    80
    %
  </p>
</Fragment>
`;
